File: Go/person.go
package main

import "fmt"

type Person struct {
	name string
	age int
}

func (p Person) info() {
	fmt.Printf("%s is %d.\n", p.name, p.age)
}

func main() {
	p := Person{name: "Alla", age: 30}
	fmt.Println(p) // {Alla 30}
	p.info() // Alla is 30.
}

File: Zig/person.zig
const warn = @import("std").debug.warn;

const Person = struct {
    name: []const u8,
    age: i32,

    pub fn info(self: Person) void {
        warn("{} is {} years old.\n", self.name, self.age);
    }
};

pub fn main() void {
    const p = Person {
        .name = "John",
        .age = 42,
    };

    p.info(); // John is 42 years old.
}

